### ISSUE TITLE
Change Boost min version to 1.66

### ISSUE THREAD

**Issue by ghost** [anura-engine/anura#285]
Spawns from https://github.com/anura-engine/anura/issues/283.

**TL;DR:** Boost `1.66.0` introduced a new header deprecating another header Anura uses. Boost `1.68.0` removes that deprecated header and Anura doesn't build anymore.

<hr />

What about doing this?

1. Upgrade Boost to `1.66.0` effectively for GNU/Linux, *BSD, etc., except MacOS and Microsoft.
    1. Recommend `1.66.0` as the minimum Boost version in the INSTALL documentation.
    2. Change path to the moved header when no `__APPLE__` and no `_MSC_VER`.
3. Upgrade Boost to `1.66.0` for MacOS and Microsoft in a calm pace as possible.

* If some platform breaks, branch off feature branch, revert changes in the `trunk` branch, finish everything before merging back.

<hr />

Stuff I don't know:

* Would this break any Steam build?

<hr />

Please confirm or reject, and comment as you wish. Thank you!

**Comment by galegosimpatico** [comment]
This can probably be unpinned once https://github.com/anura-engine/anura/issues/303 is solved.

(That one is important enough it could be pinned while it isn't fixed).

**Comment by galegosimpatico** [comment]
You can unpin and close this, now that every major platform is working on contemporary Boost (~1.70, >1.66).

However, be aware of https://github.com/anura-engine/anura/pull/305#issuecomment-667573137 in order to keep the instructions updated: https://github.com/anura-engine/anura/blob/ba8a682c96dbc816a16c30e67711794ea01a2397/INSTALL#L12-L16
